### ISSUE TITLE
Version 2 release

### ISSUE THREAD

**Issue by amattar** [elsa-workflows/elsa-core#347]
Just wondering, any plans when version 2 shall be released?

**Comment by brightermatter** [comment]
We were also wondering about the 2.0 timeline, we’d be grateful if anyone can shed some light on this.

**Comment by mohdali** [comment]
I also would like to know the timeline for Elsa 2 release.

**Comment by sfmskywalker** [comment]
Hi everyone, thank you for your interest in the V2 release!

Unfortunately I don't have a satisfying answer. Due to other priorities, I had to put a hold on working on V2 for the time being. I'm hopeful I'll be able to pickup the pace in a few months from now, but then it might take a few more months more to finish.
So realistically, I wouldn't expect V2 to be ready before the end of this year unless I'm very lucky and I get to shift focus on this earlier.

Main pieces left before V2 is ready to use:

1. Serialization needs to be fixed. Right now it's a bit of a mess.
1. Workflow Server (with REST & GraphQL endpoints to manage workflows)
1. Designer & Dashboard updates.

Originally I had planned to do a complete retake on the designer + dashboard, but I'm realizing that maybe I should do that as part of a follow-up update and not as a big bang release.

Sorry I couldn't give any better news. But I will try and see if there isn't a way I can release smaller bits so that we can at least use the improved code-first APIs, Workflow Server & multi-server support.

**Comment by amattar** [comment]
Hi Sipke,

Thanks for the prompt update.

I believe it would be beneficial to have an intermediate version, say 1.5, and release whatever ready or needs less time to implement.

I think something like multi-server support is very crucial for many as I read other comments on other issues.

**Comment by mattbab** [comment]
@sfmskywalker Are the tasks to be completed for v2 documented in issues so the community can help you release sooner? That is assuming you're open to others contributing to the project.

**Comment by sfmskywalker** [comment]
An excellent question @mattbab . I didn't have a good overview for what I consider to be absolutely necessary for Elsa 2, but your question prompted me to get a start with that and collected them under a new [Elsa 2.0 MVP](https://github.com/elsa-workflows/elsa-core/milestone/8) milestone.

Help from the community is always more than welcome, and I invite anyone who would like to help to reach out to me for questions and discussion as needed.

Please take a look, and let me know what you think. Please keep in mind these are initial drafts, and we can refine them as necessary.

In summary, here's what I think needs to be done at a minimum:

[] Reimplement serialization
[] Finish GraphQL endpoint
[] Refactor designer
[] Implement standalone client-side SPA dashboard (that communicates with a configurable Elsa GraphQL endpoint)

With that in place, Elsa 2.0 will be on-par with Elsa 1 in terms of having a dashboard + designer. But with better code-first support, and a strong foundation to build the other features.
